contracts/xsgd_contract.scilla

```diff
@@ -385,31 +385,71 @@ transition lawEnforcementWipingBurn(address : ByStr20, initiator : ByStr20)
     end
   end
 end
-transition approve(spender : ByStr20, value : Uint128, initiator : ByStr20)
-    callerIsProxy = builtin eq _sender proxy_address;
-    match callerIsProxy with
+transition increaseAllowance(spender : ByStr20, value : Uint128, initiator : ByStr20)
+  callerIsProxy = builtin eq _sender proxy_address;
+  match callerIsProxy with
+  | False =>
+    e = { _eventname : "increaseAllowance FailedNotProxy"; spender : spender; value : value; sender : initiator};
+    event e
+  | True  =>
+    isPaused <- paused;
+    isBlacklistedInitiator <- exists blacklisted[initiator];
+    isBlacklistedSpender   <- exists blacklisted[spender];
+    or_res = orb_3 isPaused isBlacklistedInitiator isBlacklistedSpender;
+    match or_res with
     | False =>
-      e = { _eventname : "approve FailedNotProxy"; spender : spender; value : value; sender : initiator};
+      get_current_allowance <- allowed[initiator][spender];
+      current_allowance = 
+        match get_current_allowance with
+        | Some allowance => allowance
+        | None => zero
+        end;
+      new_allowance = builtin add current_allowance value;
+      allowed[initiator][spender] := new_allowance;
+      e = {_eventname : "IncreasedAllowance"; sender : initiator; spender : spender; new_allowance : new_allowance};
+      event e;
+      msg = {_tag : "IncreaseAllowanceSuccessCallBack"; _recipient : initiator; _amount : zero;
+              sender : initiator; spender : spender; amount : value };
```

Comment:
This might be a stupid question, but is there any reason to have the `sender` field in this message? Its value is the same as the value of `_recipient`, which means that whoever receives this callback already knows that they are the `sender`.